src/service/onesignal/index.ts

```diff
@@ -30,6 +30,8 @@ const rebuildAndSubscribeTags = async () => {
 
         resolve(null);
       }, SET_TAG_TIMEOUT);
+
+      OneSignal.User.addTag('has_user_posted', hasUserPosted ? 'true' : 'false');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Adding the 'has_user_posted' tag based on the `hasUserPosted` value is a good approach. However, it's important to handle any potential errors that could occur during this operation. Consider wrapping this line in a try-catch block to handle any exceptions that may arise.

```javascript
try {
  OneSignal.User.addTag('has_user_posted', hasUserPosted ? 'true' : 'false');
} catch (e) {
  console.error('Error adding has_user_posted tag: ', e);
}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->